### PR TITLE
Fix Views and Serializers

### PR DESCRIPTION
**Pull request by NyanHelsing**

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -1,41 +1,50 @@
...
+# Views
+# #############################################################################
 
 
-class UserSearchView(HaystackViewSet):
-    index_models = [User]
-    serializer_class = UserSearchSerializer
-
-
-class LargeResultsSetPagination(pagination.PageNumberPagination):
```
Comment (cameronblandford) [inline api/views.py]: Why did you delete this? Removing it would break how pagination against the items endpoint is handled on the frontend.

File: api/views.py [hunk 2]
```diff
@@ -44,517 +53,157 @@ def api_root(request):
...
-    they will be "pending".
-
-    #This Request/Response
+    # `ItemViewSet`
     """
+
+    queryset = Item.objects.all()
     serializer_class = ItemSerializer
-    permission_classes = (drf_permissions.IsAuthenticatedOrReadOnly, )
-    pagination_class = LargeResultsSetPagination
```
Comment (cameronblandford) [inline api/views.py]: Ditto for this, this is why we don't have to make 10 separate requests to populate the schedule

File: api/views.py [hunk 3]
```diff
@@ -44,517 +53,157 @@ def api_root(request):
...
-            Success:       200 OK + item representation
+        collection = Collection.objects.get(self.request.data.get("collection"))
+        if not user.has_perm('collection.AddItem', collection):
+            return HttpResponse('Unauthorized', status=401)
+        serializer.save()
 
-    Note: The `title`, `type`, `status` and `category` fields are required for PUT and optional for PATCH requests.
+    def retrieve(self, request, *args, **kwargs):
+        instance = self.get_object()
+        #if request.user.has_perm("view", instance):
```
Comment (cameronblandford) [inline api/views.py:158]: Remove commented code before merge

File: api/views.py [hunk 4]
```diff
@@ -44,517 +53,157 @@ def api_root(request):
...
-        username                      string                  username for the user
-        first_name                    string                  first name of the user
-        last_name                     string                  last name of the user
-        email                         string                  email address associated with the user's account
-        date_joined                   iso8601 timestamp       date/time of account creation
-        last_login                    iso8601 timestamp       date/time of last login
-        is_active                     boolean                 whether the user account is active
-        gravatar                      string                  link to user gravatar
-        token                         string                  access token for social account (used for OAUTH)
+# Search Views
```
Comment (cameronblandford) [inline api/views.py]: :+1: nice separation

File: service/settings/base.py [hunk 5]
```diff
@@ -144,7 +144,7 @@
         'rest_framework.parsers.MultiPartParser'
     ),
     'DEFAULT_RENDERER_CLASSES': (
-        'api.renderers.CustomJSONRenderer',
+        'rest_framework_json_api.renderers.JSONRenderer',
```
Comment (cameronblandford) [inline service/settings/base.py:147]: :+1::+1::+1::+1:

File: api/views.py (hunk 3 above)
Comment (NyanHelsing) [reply api/views.py:158]: This is getting uncommented in the next pr

**Comment by cameronblandford** [comment]
looks good 👌
